drivers/include/mtdi.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2016  OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_mtdi Memory Technology Device
+ * @ingroup     drivers
+ * @brief       Low level Memory Technology Device interface
+ *
+ * Generic memory technology device interface
+ *
+ * @{
+ *
+ * @file
+ *
+ * @author      Aurelien Gonce <aurelien.gonce@altran.com>
+ */
+
+#ifndef __MTDI_H_
+#define __MTDI_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Results of Memory Functions */
+typedef enum {
+    MTDI_RES_OK = 0,     /**< 0: Successful */
+    MTDI_RES_ERROR,      /**< 1: R/W Error */
+    MTDI_RES_WRPRT,      /**< 2: Write Protected */
+    MTDI_RES_NOTRDY,     /**< 3: Not Ready */
+    MTDI_RES_PARERR      /**< 4: Invalid Parameter */
+} mtdi_result_t;
+
+/**
+ * @name Disk Status Bits
+ * @{
+ */
+typedef enum {
+    MTDI_STA_INIT    = 0x00, /**< Drive initialized */
+    MTDI_STA_NOINIT  = 0x01, /**< Drive not initialized */
```

Comment:
IMO it's better to have noinit = 0, because then the initial value will be set to "noinit" on boot even if placed in .bss